### PR TITLE
Adds an alternative explosive appendicitis

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

When someone gains appendicitis, there's a 35% chance it will be EXPLOSIVE appendicitis.

This version inmediately appears as the red 'DANGER' on medical HUDs.

It takes significantly longer to reach the end-stage than normal appendicitis. It has more stages and is 50% slower at advancing them.

Stabilizing Agent will clear the inflammation if administered before Stage 3.

Once the appendicitis reaches end-stage, a fuse sound will play and the unfortunate victim will blow up! Stabilizing Agent, again, will help. It will delay the ignition for as long as it's in the system.

When the appendix is surgically removed, it will similarly 'prime' and become ready to detonate in a few seconds, though it's shoved into the surgeon's hands.

Refactored dangerous surgical removal to be a bespoke element with the aforementioned sparking effects and delayed explosion.

## Why It's Good For The Game

Appendicitis is kinda lame and boring honestly. It's just 'oof ouch my appendix let's hope medbay wasnt blown up yet'.

This new appendicitis makes getting it much scarier! It will literally blow up if left unchecked. This adds panic and fretting, with a really funny final stage, which also technically makes it self-treatable if you do something silly like grab a bomb suit and some atropine before it blows. If there's no medics available you can indefinitely stall appendicitis with Stabilizing Agent if it's the explosive version, or the new medicine for the appendix if not. The latter works with reduced efficiency as well, for the explosive version.

Being in the OR room and suddenly getting a frickin' detonating organ put in your hand is gonna cause some really funny moments, especially if the surgery is being recklessly performed in the general treatment center. Same with surplus prosthetics getting this treatment, but in that case it will be a lot more harmless.

Basically it turns a boring 'ok time to get surgery' event into a 'oh shit i'm about to fucking explode' event. It's a 35% chance so that it doesn't get boring fast.

## Changelog

:cl:
add: When someone gains appendicitis, there's a 35% chance it will be EXPLOSIVE appendicitis.
add: This version inmediately appears as the red 'DANGER' on medical HUDs.
add: It takes significantly longer to reach the end-stage than normal appendicitis. It has more stages and is 50% slower at advancing them.
add: Stabilizing Agent will clear the inflammation if administered before Stage 3.
add: Once the appendicitis reaches end-stage, a fuse sound will play and the unfortunate victim will blow up! Stabilizing Agent, again, will help. It will delay the ignition for as long as it's in the system.
add: When the appendix is surgically removed, it will similarly 'prime' and become ready to detonate in a few seconds, though it's shoved into the surgeon's hands.
refactor: Refactored dangerous surgical removal to be a bespoke element with the aforementioned sparking effects and delayed explosion, which now applies to surgical prostheses as well.
/:cl:

### CODE REVIEW

**Comment by EuSouAFazer** [comment]
Turns out the medical community was wrong for centuries - the appendicy wasn't useless, but rather, a secret bomb to be deployed at a tactical moment. The moment just so happens to be in a space station, 30 billion light years away from Earth and with a lizard clown called Eats-The-Appendix chasing after you with a knife.

**Comment by carlarctg** [comment]
https://www.youtube.com/watch?v=TrZESCp2IsE

**Comment by EricZilla** [comment]
What if we... what if we made it so pulling out an organ always puts it in a free hand if one is available

**Comment by SomeoneYouProbablyKnow** [comment]
Amusing but considering 50% of players don't do jack about regular appendicitis they might just explode without knowing what hit em. 

Give em a low beeping sound. That should help em.

**Comment by JohnFulpWillard** [comment]
I find this pr funny but you still have to fill out why it's good for the game

**Comment by carlarctg** [comment]
1984

**Comment by carlarctg** [comment]
dopne

File: code/datums/elements/dangerous_surgical_removal.dm [hunk 1]
```diff
@@ -5,25 +5,45 @@
  * That's about it.
  */
 /datum/element/dangerous_surgical_removal
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	/// If not null, can snap cable restraints and similar.
```
Comment (ATH1909) [inline code/datums/elements/dangerous_surgical_removal.dm:10]: ??????????????????

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -1459,6 +1459,21 @@
...
+	self_consuming = FALSE
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+
+/datum/reagent/medicine/lamphrite/on_mob_metabolize(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob, TRAIT_STABLEAPPENDIX, type)
+
+/datum/reagent/medicine/lamphrite/on_mob_end_metabolize(mob/living/affected_mob)
+	. = ..()
+	REMOVE_TRAIT(affected_mob, TRAIT_STABLEAPPENDIX, type)
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1475]: I thought stabilizing agent was supposed to have this trait, not some new chem?

Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1475]: You misunderstood, read the post again

File: code/datums/elements/dangerous_surgical_removal.dm (hunk 1 above)
Comment (carlarctg) [reply code/datums/elements/dangerous_surgical_removal.dm:10]: i pasted it from that element

```suggestion
```

**Comment by Wallemations** [comment]
35% sounds like a LOT

**Comment by SomeoneYouProbablyKnow** [comment]
Appendicitis is not a fun disease to get. Cutting it down to 65% is a positive.

**Comment by Wallemations** [comment]
> Appendicitis is not a fun disease to get. Cutting it down to 65% is a positive.

This doesn't cut it down though, this just makes a variant that makes you EXPLODE.
If anything that's way worse, I imagine an explosion on the same tile as you is gonna gib you.

**Comment by carlarctg** [comment]
No it's not. It just knocks you into crit and maybe makes you lose a limb. It's balanced out by being completely self treatable.

File: code/__DEFINES/traits/declarations.dm [hunk 3]
```diff
@@ -107,6 +107,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 /// Prevents you from leaving your corpse
 #define TRAIT_CORPSELOCKED "corpselocked"
 #define TRAIT_STABLELIVER "stable_liver"
+#define TRAIT_STABLEAPPENDIX "stable_appendix"
```
Comment (Mothblocks) [inline code/__DEFINES/traits/declarations.dm:110]: `STABLE_APPENDIX`--the others shouldn't be like this

File: code/datums/elements/dangerous_surgical_removal.dm [hunk 4]
```diff
@@ -5,25 +5,44 @@
  * That's about it.
  */
 /datum/element/dangerous_surgical_removal
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	var/fuse_time = 3 SECONDS
+	/// If not null, can snap handcuffs.
+	var/explosion_strength = 1
+	var/flag_blockers = (ORGAN_FAILING|ORGAN_EMP)
```
Comment (Mothblocks) [inline code/datums/elements/dangerous_surgical_removal.dm:13]: Document the others

File: code/datums/elements/dangerous_surgical_removal.dm [hunk 5]
```diff
@@ -5,25 +5,44 @@
...
-	qdel(source)
+	playsound(source, 'sound/effects/fuse.ogg', 45)
+	var/mutable_appearance/sparks = mutable_appearance('icons/effects/welding_effect.dmi', "welding_sparks", GASFIRE_LAYER, source, ABOVE_LIGHTING_PLANE)
+	source.add_overlay(sparks)
+	LAZYADD(source.update_overlays_on_z, sparks)
+	source.visible_message(span_userdanger("As [user] pulls [source] from [old_owner]'s body, it begins making a concerning sizzling noise..."))
+	animate(source, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
+	animate(time = 1, pixel_z = 0, easing = BOUNCE_EASING)
+	for(var/i in 1 to 32)
+		animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING)
```
Comment (Mothblocks) [inline code/datums/elements/dangerous_surgical_removal.dm:42]: Use existing color defines for this

File: code/datums/elements/dangerous_surgical_removal.dm [hunk 6]
```diff
@@ -5,25 +5,44 @@
...
+	if(source.organ_flags & flag_blockers)
 		return
-	if(user?.Adjacent(source))
-		source.audible_message("[source] explodes on [user]'s face!")
-		user.take_bodypart_damage(15)
-	else
-		source.audible_message("[source] explodes into tiny pieces!")
-	explosion(source, light_impact_range = 1, explosion_cause = source)
-	qdel(source)
+	playsound(source, 'sound/effects/fuse.ogg', 45)
```
Comment (Mothblocks) [inline code/datums/elements/dangerous_surgical_removal.dm:34]: `vol = 45`

File: code/datums/elements/dangerous_surgical_removal.dm [hunk 7]
```diff
@@ -5,25 +5,44 @@
...
+	LAZYADD(source.update_overlays_on_z, sparks)
+	source.visible_message(span_userdanger("As [user] pulls [source] from [old_owner]'s body, it begins making a concerning sizzling noise..."))
+	animate(source, time = 1, pixel_z = 12, easing = ELASTIC_EASING)
+	animate(time = 1, pixel_z = 0, easing = BOUNCE_EASING)
+	for(var/i in 1 to 32)
+		animate(color = (i % 2) ? "#ffffff": "#ff6739", time = 1, easing = QUAD_EASING)
+
+	ASYNC
+		user.put_in_hands(source, del_on_fail = FALSE, forced = TRUE)
+		sleep(fuse_time)
```
Comment (Mothblocks) [inline code/datums/elements/dangerous_surgical_removal.dm:46]: Use `stoplag` instead

File: code/modules/surgery/organs/internal/appendix/_appendix.dm [hunk 8]
```diff
@@ -72,19 +107,76 @@
 				organ_owner.vomit(VOMIT_CATEGORY_DEFAULT, lost_nutrition = 95)
 				organ_owner.adjustOrganLoss(ORGAN_SLOT_APPENDIX, 15)
 
+/obj/item/organ/internal/appendix/proc/get_stabilized_status()
+	var/list/holder_list = owner?.reagents.reagent_list
+	if(locate(/datum/reagent/stabilizing_agent) in holder_list)
+		return TRUE
+	return FALSE
```
Comment (Mothblocks) [inline code/modules/surgery/organs/internal/appendix/_appendix.dm:114]: Cleaner to return !isnull

File: code/modules/surgery/organs/internal/appendix/_appendix.dm [hunk 9]
```diff
@@ -72,19 +107,76 @@
...
+			else if(SPT_PROB(2.5, seconds_per_tick))
+				INVOKE_ASYNC(organ_owner, TYPE_PROC_REF(/mob, emote), "sway")
+				to_chat(organ_owner, span_warning("You're becoming delirious from the explosive pain in your chest."))
+				organ_owner.set_eye_blur_if_lower(6 SECONDS * seconds_per_tick)
+		if(7)
+			INVOKE_ASYNC(src, PROC_REF(kaboom))
+			inflamation_stage = EXPLOSIVE_MAX_STAGE // nothing happens in 8
+
+/obj/item/organ/internal/appendix/proc/kaboom()
+	sleep(60 SECONDS)
```
Comment (Mothblocks) [inline code/modules/surgery/organs/internal/appendix/_appendix.dm:159]: `stoplag`

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request closed without merging**